Created gradle structure and build file

mihnita opened the pull request:
Also disabled the failing test case.

* Still using JUnit 3 style (no `@RunWith` and `@Test`, and still extending `TestCase`)
  Wasn't sure if you want to move to JUnit 4. Cleaner to be a separate change anyway.
* gradle javadoc builds the documentation
* gradle build and gradle install create the `kxml2-2.3.1-sources.jar`
  The naming convention is different from the old one, but more "standard"
   (`kxml2-2.3.1-sources.jar` instead of `kxml2-src-2.3.1.zip`)
* Not building the min version (yet?)

stefanhaustein commented on build.gradle:
```diff
@@ -0,0 +1,30 @@
+apply plugin: 'maven'
+
+group = 'net.sf.kxml'
+version = '2.3.1'
```
Should be the latest tag (+1?)

mihnita replied:
I am still figuring out how to update an existing pull request :-)
Can you please update it?

stefanhaustein replied:
Committ to your branch and do a push, and the pull request will get updated automatically.

stefanhaustein commented on build.gradle:
```diff
@@ -0,0 +1,30 @@
+group = 'net.sf.kxml'
+version = '2.3.1'
+
+sourceCompatibility = 1.7
```
Should be lower for CLDC compat.

mihnita replied:
Sorry, I don't know much (anything :-) about CLDC.
Please feel free to change the version to whatever is needed.

What I was able to find was this:
  "CLDC 8 is a subset of Java SE 8 that incorporates the language, API, and VM updates."
  (http://docs.oracle.com/javame/config/cldc/opt-pkgs/api/cldc/api/index.html)

If there is a way to check compatibility automatically, it would be best :-)

stefanhaustein replied:
Wasn't aware there is a new version... I think we should keep compatibility with CLDC 1.1 until a major version bump. On wikipedia, I found:  "... all Java ME platforms are currently restricted to JRE 1.3 features and use that version of the class file format (internally known as version 47.0)".

mihnita commented:
Figured out how to update a pull request :-)

Updated version to 2.3.2

Updated Java version to 1.2
This seems crazy old, please change as you wish, or let me know what version to use.
I went with 1.2 based on the version of the .class files in the [last public release in maven (2.3.0)](https://mvnrepository.com/artifact/net.sf.kxml/kxml2/2.3.0)

`javap -v -cp kxml2-2.3.0.jar org.kxml2.io.KXmlParser | grep version`

The results is
```
  minor version: 0
  major version: 46
```
and that matches Java 1.2

stefanhaustein commented on build.gradle:
```diff
@@ -0,0 +1,30 @@
+apply plugin: 'maven'
+
+group = 'net.sf.kxml'
+version = '2.3.2'
```
I think it should be at least 2.4.2, as I have already tagged the last bugfix as 2.4.1 (given the project move and time span -- perhaps a bit early): https://github.com/stefanhaustein/kxml2/releases

In my view, a 2.5.0. would be better for a new project layout.

mihnita commented:
Updated library version to 2.5.0 and JKD version to 1.3
